client/dashboard/me/security-account-recovery/recovery-sms.tsx

```diff
@@ -0,0 +1,280 @@
+import {
+	accountRecoveryQuery,
+	updateAccountRecoverySMSMutation,
+	removeAccountRecoverySMSMutation,
+	resendAccountRecoverySMSValidationMutation,
+	validateAccountRecoverySMSCodeMutation,
+} from '@automattic/api-queries';
+import { useMutation, useSuspenseQuery } from '@tanstack/react-query';
+import {
+	__experimentalVStack as VStack,
+	__experimentalConfirmDialog as ConfirmDialog,
+	__experimentalInputControl as InputControl,
+	Button,
+	Card,
+	CardBody,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm } from '@wordpress/dataviews';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useMemo, useState } from 'react';
+import { ButtonStack } from '../../components/button-stack';
+import Notice from '../../components/notice';
+import { SectionHeader } from '../../components/section-header';
+import { validatePhone } from '../../utils/phone-number';
+import PhoneNumberInput from './phone-number-input';
+import type { SecuritySMSFormData } from './types';
+import type { Field } from '@wordpress/dataviews';
+
+const initialFormData: SecuritySMSFormData = {
+	smsNumber: {
+		countryCode: '',
+		phoneNumber: '',
+		countryNumericCode: '',
+	},
+	smsCode: '',
+};
+
+export default function RecoverySMS() {
+	const { data: accountRecoveryData } = useSuspenseQuery( accountRecoveryQuery() );
+
+	const { mutate: validateSMS, isPending: isValidateSMSPending } = useMutation(
+		updateAccountRecoverySMSMutation()
+	);
+	const { mutate: removeSMS, isPending: isRemoveSMSPending } = useMutation(
+		removeAccountRecoverySMSMutation()
+	);
+	const { mutate: resendValidation, isPending: isResendValidationPending } = useMutation(
+		resendAccountRecoverySMSValidationMutation()
+	);
+	const { mutate: validateSMSCode, isPending: isValidateSMSCodePending } = useMutation(
+		validateAccountRecoverySMSCodeMutation()
+	);
+
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const accountRecoveryPhone = accountRecoveryData.phone;
+
+	const [ formData, setFormData ] = useState< SecuritySMSFormData >( {
+		...initialFormData,
+		smsNumber: {
+			phoneNumber: accountRecoveryPhone?.number || '',
+			countryCode: accountRecoveryPhone?.country_code || '',
+			countryNumericCode: accountRecoveryPhone?.country_numeric_code || '',
+		},
```

Comment:
[nit] I think we don't need `initialFormData` as the shape is completely defined here:

```suggestion
		smsNumber: {
			phoneNumber: accountRecoveryPhone?.number || '',
			countryCode: accountRecoveryPhone?.country_code || '',
			countryNumericCode: accountRecoveryPhone?.country_numeric_code || '',
		},
		smsCode: '',
```

Reply:
Updated it here: https://github.com/Automattic/wp-calypso/pull/105530/commits/ecbff47f14d016f8077767e45a686d94a22257eb